Formula/v/vulkan-loader.rb

```diff
@@ -34,8 +34,12 @@ class VulkanLoader < Formula
   end
 
   def install
+    inreplace "loader/vulkan.pc.in" do |s|
+      s.gsub!(/includedir=.*/, "includedir=#{Formula["vulkan-headers"].include}")
+    end
```

Comment:
Is there an upstream issue for this?

Reply:
I'm not sure this is really an _issue_ upstream per se; they just expect that headers will be installed with the same prefix as the loader (which is true in traditional linux package management). In theory they could probably format this into the .pc themselves, but I could see that causing issues for some other cases.

Reply:
How about asking them to make the header install location configurable?

Reply:
Hmmm, it looks like it _used_ to be (since https://github.com/KhronosGroup/Vulkan-Loader/commit/d9faa0f8442d13fff3041f92c299a6361a4ebf41), but that was removed in https://github.com/KhronosGroup/Vulkan-Loader/commit/520eaa5a525adce0cdbdc7b8c7f7d8f0375340a6 to address https://github.com/KhronosGroup/Vulkan-Loader/issues/1318 and https://github.com/KhronosGroup/Vulkan-Loader/issues/1319 (BSD/Nix stuff).

We used to have an `inreplace` like this, but it was removed in 498d5de38c74b34a24b484abc75e38a26149d38f (#150267) for reasons I'm not clear on. There's currently an open issue for this (#152107).

Reply:
Oh, wait, no, this was never actually supported upstream. Hmm. Though I'm pretty sure that if they _did_ add it, it'd break @jbeich's use-case (which involves building to a tmp dir), at least initially.

Reply:
Could you make an issue asking them to add that choice back? Then I'd be fine with the inreplace until they do.

Reply:
Vulkan-Loader can be built with **bundled** Vulkan-Headers. It can be used to update Vulkan-Headers independently, when Vulkan-Loader fails to build with the new version.

Reply:
Hmm, maybe `vulkan-headers` shouldn't be its own formula, and instead `vulkan-loader` (renamed to `vulkan`?) should pull the headers in as a resource?

Reply:
In any case, filed https://github.com/KhronosGroup/Vulkan-Loader/issues/1359 for now.

Reply:
Specifying `CMAKE_INSTALL_FULL_INCLUDEDIR` should work based on what I am seeing here. You will also need to specify `CMAKE_INSTALL_FULL_LIBDIR` and ensure CMAKE_INSTALL_PREFIX evaluates to an empty string.

```
    # BUG: The following code will NOT work well with `cmake --install ... --prefix <dir>`
    # due to this code relying on CMAKE_INSTALL_PREFIX being defined at configure time.
    #
    # NOTE: vulkan.pc essentially cover both Vulkan-Loader and Vulkan-Headers for legacy reasons.
    if ("${CMAKE_INSTALL_PREFIX}" STREQUAL "")
        set(CMAKE_INSTALL_LIBDIR_PC ${CMAKE_INSTALL_FULL_LIBDIR})
        set(CMAKE_INSTALL_INCLUDEDIR_PC ${CMAKE_INSTALL_FULL_INCLUDEDIR})
    else()
        file(RELATIVE_PATH CMAKE_INSTALL_LIBDIR_PC ${CMAKE_INSTALL_PREFIX} ${CMAKE_INSTALL_FULL_LIBDIR})
        file(RELATIVE_PATH CMAKE_INSTALL_INCLUDEDIR_PC ${CMAKE_INSTALL_PREFIX} ${CMAKE_INSTALL_FULL_INCLUDEDIR})
    endif()
```

Reply:
Done.

Reply:
make sense, yeah, I removed `inreplace` mostly because of the pc patch is not usable (I think)